src/ripple/overlay/impl/PeerImp.cpp

```diff
@@ -653,6 +653,23 @@ void
 PeerImp::doProtocolStart()
 {
     onReadMessage(error_code(), 0);
+
+    protocol::TMManifests tm;
+    tm.set_history (true);
+
+    overlay_.manifestCache ().for_each_manifest (
```

Comment:
@nbougalis It's an interesting thought. If we cache it we'd still need to hold a lock when we copied the cached `protocol::TMManifest`. So the questions I need to resolve are:

1. Are other threads likely to block on this lock?
2. How expensive is it to create the `TMManifest` each time? If it's cheap it won't be a bottleneck and we don't care.
3. How much less expensive is it to copy the `TMManifest`?
4. We would need to hold the lock slightly longer in `applyManifest` when we update the cached `TMManifest`.
5. Is it worth the extra code complexity for the trade-offs in (3) and (4).

One difference between creating and copying manifests may be memory allocations. I'll look into this.